src/main/java/frc/robot/subsystems/drivetrain/Chassis.kt

```diff
@@ -81,6 +98,46 @@ object Chassis :
     private val steerCharacterization = SwerveRequest.SysIdSwerveSteerGains()
     private val rotationCharacterization = SwerveRequest.SysIdSwerveRotation()
 
+    private val pathApplyRobotSpeeds = ApplyRobotSpeeds()
+
+    fun configureAutoBuilder() {
+        try {
+            val config = RobotConfig.fromGUISettings()
+            AutoBuilder.configure(
+                { state.Pose }, // Supplier of current robot pose
+                this::resetPose, // Consumer for seeding pose against auto
+                { state.Speeds }, // Supplier of current robot speeds
+                // Consumer of ChassisSpeeds and feedforwards to drive the robot
+                { speeds: ChassisSpeeds, feedforwards: DriveFeedforwards ->
+                    Chassis.setControl(
+                        pathApplyRobotSpeeds
+                            .withSpeeds(speeds)
+                            .withWheelForceFeedforwardsX(feedforwards.robotRelativeForcesXNewtons())
+                            .withWheelForceFeedforwardsY(feedforwards.robotRelativeForcesYNewtons())
+                    )
+                },
+                PPHolonomicDriveController( // PID constants for translation
+                    PIDConstants(10.0, 0.0, 0.0), // PID constants for rotation
+                    PIDConstants(7.0, 0.0, 0.0),
+                ),
+                config, // Assume the path needs to be flipped for Red vs Blue, this is normally the
+                // case
+                { DriverStation.getAlliance().orElse(Alliance.Blue) == Alliance.Red },
```

Comment:
The field is rotationally symmetric this year, so we don't want to flip when we're on red